course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/EventsActor.java

```diff
@@ -374,10 +374,27 @@ private void eventEnroll(Request request) throws Throwable {
         }
     }
 
-    private void validateEnrolment(EventBatch batchData, UserEvents enrolmentData, Boolean isEnrol){
+    private void validateEnrolment(EventBatch batchData, UserEvents enrolmentData, Boolean isEnrol,Map<String, Object> contentDetails){
+        String endDateStr = (String) contentDetails.get("endDate");
+        String endTimeStr = (String) contentDetails.get("endTime");
+
+        LocalDate eventEndDate = LocalDate.parse(endDateStr);
+
+        OffsetTime endTime = OffsetTime.parse(endTimeStr);
+
+        LocalDateTime eventEndLocalDateTime = LocalDateTime.of(eventEndDate, endTime.toLocalTime());
+
+        ZonedDateTime eventEndDateTime = eventEndLocalDateTime.atZone(ZoneId.systemDefault()).withZoneSameInstant(endTime.getOffset());
+
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.systemDefault());
+
+        if (now.isAfter(eventEndDateTime)) {
+            ProjectCommonException.throwClientErrorException(ResponseCode.eventEnded,
+                    ResponseCode.eventEnded.getErrorMessage());
+        }
```

Comment:
this is same as lines 410, 416
either these mentioned lines should be replaced by this or keep this and remove those mentioned lines